pkg-manager/core/src/install/index.ts

```diff
@@ -407,30 +414,68 @@ export async function mutateModules (
         injectWorkspacePackages: opts.injectWorkspacePackages,
       }
     }
-    if (
-      !ctx.lockfileHadConflicts &&
-      !opts.fixLockfile &&
-      !opts.dedupe &&
-      installsOnly &&
-      (
-        frozenLockfile ||
-        opts.ignorePackageManifest ||
-        !needsFullResolution &&
-        opts.preferFrozenLockfile &&
-        (!opts.pruneLockfileImporters || Object.keys(ctx.wantedLockfile.importers).length === Object.keys(ctx.projects).length) &&
-        ctx.existsNonEmptyWantedLockfile &&
-        ctx.wantedLockfile.lockfileVersion === LOCKFILE_VERSION &&
-        await allProjectsAreUpToDate(Object.values(ctx.projects), {
-          catalogs: opts.catalogs,
-          autoInstallPeers: opts.autoInstallPeers,
-          excludeLinksFromLockfile: opts.excludeLinksFromLockfile,
-          linkWorkspacePackages: opts.linkWorkspacePackagesDepth >= 0,
-          wantedLockfile: ctx.wantedLockfile,
-          workspacePackages: ctx.workspacePackages,
-          lockfileDir: opts.lockfileDir,
-        })
-      )
-    ) {
+
+    /**
+     * Attempt to perform a "frozen install".
```

Comment:
I wrote this comment after manually reading this function to see what it does. Let me know if anything in here is inaccurate.